test/integration/build_to_git_test.go

```diff
@@ -0,0 +1,244 @@
+
+			// wait until the Build finish the validation
+			buildObject, err := tb.GetBuildTillRegistration(buildName, corev1.ConditionFalse)
+			Expect(err).To(BeNil())
+			// this one is validating file protocol
+			Expect(buildObject.Status.Registered).To(Equal(corev1.ConditionFalse))
+			Expect(buildObject.Status.Reason).To(Equal("invalid source url"))
+		})
+	})
+
+	Context("when a build reference a invalid repository with a false annotation for sourceURL", func() {
+		It("should not validate sourceURL", func() {
+
+			// populate Build related vars
+			buildName := BUILD + tb.Namespace
+			buildObject, err = tb.Catalog.LoadBuildWithNameAndStrategy(
+				buildName,
+				STRATEGY+tb.Namespace,
+				[]byte(test.BuildCBSWithVerifyRepositoryAnnotation),
+			)
+			Expect(err).To(BeNil())
+
+			buildObject.ObjectMeta.Annotations["build.build.dev/verify.repository"] = "false"
+			buildObject.Spec.Source.URL = "foobar"
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			// wait until the Build finish the validation
+			buildObject, err := tb.GetBuildTillRegistration(buildName, corev1.ConditionTrue)
+			Expect(err).To(BeNil())
+			// skip validation due to false annotation
+			Expect(buildObject.Status.Registered).To(Equal(corev1.ConditionTrue))
+			Expect(buildObject.Status.Reason).To(Equal("Succeeded"))
+		})
+	})
+
+	Context("when the build source url protocol is https plus github enterprise and the verify annotation is true", func() {
+		It("should fail validating source url", func() {
+
+			// populate Build related vars
+			buildName := BUILD + tb.Namespace
+			buildObject, err = tb.Catalog.LoadBuildWithNameAndStrategy(
+				buildName,
+				STRATEGY+tb.Namespace,
+				[]byte(test.BuildCBSWithVerifyRepositoryAnnotation),
+			)
+			Expect(err).To(BeNil())
+
+			buildObject.ObjectMeta.Annotations["build.build.dev/verify.repository"] = "true"
+			buildObject.Spec.Source.URL = "https://github.ibm.com/coligo/build-fake"
+
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			// wait until the Build finish the validation
+			buildObject, err := tb.GetBuildTillRegistration(buildName, corev1.ConditionFalse)
+			Expect(err).To(BeNil())
+			Expect(buildObject.Status.Registered).To(Equal(corev1.ConditionFalse))
+			// Because github enterprise always require authentication, this validation will fail while
+			// the repository could not be found.
+			Expect(buildObject.Status.Reason).To(Equal("remote repository unreachable"))
+		})
+
+		It("should not validate sourceURL because a referenced secret exists", func() {
+
+			// populate Build related vars
+			buildName := BUILD + tb.Namespace
+			buildObject, err = tb.Catalog.LoadBuildWithNameAndStrategy(
+				buildName,
+				STRATEGY+tb.Namespace,
+				[]byte(test.BuildCBSWithVerifyRepositoryAnnotation),
+			)
+			Expect(err).To(BeNil())
+
+			buildObject.ObjectMeta.Annotations["build.build.dev/verify.repository"] = "true"
+			buildObject.Spec.Source.URL = "https://github.ibm.com/coligo/build-fake"
```

Comment:
Not sure how that messes up the test, but I would opt for a different example here.
```suggestion
			buildObject.Spec.Source.URL = "https://github.yourco.com/org/build-fake"
```